src/pages/index.js

```diff
@@ -0,0 +1,365 @@
+
+  const originalButtonText = editSubmitButton.textContent;
+  editSubmitButton.textContent = "Saving...";
+  editSubmitButton.disabled = true;
+
+  const updatedName = editModalNameInput.value;
+  const updatedAbout = editModalDescription.value;
+
+  api
+    .editUserInfo({ name: updatedName, about: updatedAbout })
+    .then((updatedUserInfo) => {
+      profileName.textContent = updatedUserInfo.name;
+      profileDescription.textContent = updatedUserInfo.about;
+      closeModal(editModal);
+    })
+    .catch((err) => {
+      console.error("Error updating profile:", err);
+      alert("Could not update profile. Please try again later.");
+    })
+    .finally(() => {
+      hideLoadingModal(); // Hide loading modal
+      editSubmitButton.textContent = originalButtonText;
+      editSubmitButton.disabled = false;
+    });
+});
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", closeModalOnEscape);
+  modal.addEventListener("mousedown", closeModalOnOverlay);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", closeModalOnEscape);
+  modal.removeEventListener("mousedown", closeModalOnOverlay);
+}
+
+function closeModalOnEscape(evt) {
+  if (evt.key === "Escape") {
+    const openedModal = document.querySelector(".modal_opened");
+    if (openedModal) {
+      closeModal(openedModal);
+    }
+  }
+}
+
+function closeModalOnOverlay(evt) {
+  if (evt.target.classList.contains("modal")) {
+    closeModal(evt.target);
+  }
+}
+
+function disableButton(buttonElement, config) {
+  buttonElement.classList.add(config.inactiveButtonClass);
+  buttonElement.disabled = true;
+}
+
+cardModalBtn.addEventListener("click", () => {
+  openModal(cardModal);
+});
+
+const previewModal = document.querySelector("#preview-modal");
+const previewCloseButton = previewModal.querySelector(
+  ".modal__close-button_preview"
+);
+previewCloseButton.addEventListener("click", () => closeModal(previewModal));
+
+cardModalCloseBtn.addEventListener("click", () => {
+  closeModal(cardModal);
+});
+
+cardForm.addEventListener("submit", (evt) => {
+  evt.preventDefault();
+  showLoadingModal("Creating post...");
+
+  const originalButtonText = cardSubmitBtn.textContent;
+  cardSubmitBtn.textContent = "Saving...";
+  cardSubmitBtn.disabled = true;
+
+  const inputValues = {
+    name: cardCaptionInput.value,
+    link: cardNameInput.value,
+  };
+
+  api
+    .createCard(inputValues)
+    .then((cardData) => {
+      renderCard(cardData, "prepend");
+      closeModal(cardModal);
+      cardForm.reset();
+      disableButton(cardSubmitBtn, settings);
+    })
+    .catch((err) => {
+      console.error("Error creating card:", err);
+      alert("Could not create card. Please try again later.");
+    })
+    .finally(() => {
+      hideLoadingModal();
+      cardSubmitBtn.textContent = originalButtonText;
+      cardSubmitBtn.disabled = false;
```

Comment:
`cardSubmitBtn.disabled = false;` is not needed in `.finally` because it activates the submit button, but you disable it in `.then`


You can do that in `.catch`, to enable the button if there is an error


```js
    .catch((err) => {
      console.error("Error creating card:", err);
      alert("Could not create card. Please try again later.");
      cardSubmitBtn.disabled = false;  // here
    })
    .finally(() => {
      hideLoadingModal();
      cardSubmitBtn.textContent = originalButtonText;
    });

```